### PR TITLE
Refactored fullscreen navigation

### LINKED ISSUE #292: Back button not redirecting when in full-screen mode
### Actual
back

### Expected
Back button should redirect to the edit page.

### PR DESCRIPTION
**Pull request by siddhant1**
Closes #292

### CODE REVIEW

File: src/components/renderer/renderer.tsx [hunk 1]
```diff
@@ -139,7 +140,24 @@ class Editor extends React.PureComponent<Props, State> {
...
     }
+
+    window.addEventListener(
+      'popstate',
+      event => {
+        const pathname = window.location.href;
+        const urlarray = pathname.split('/');
+        const lastword = urlarray[urlarray.length - 1];
+        event.preventDefault();
+        if (lastword === 'view') {
```
Comment (domoritz) [inline src/components/renderer/renderer.tsx]: pathname.endsWith('view')

File: src/components/renderer/renderer.tsx [hunk 2]
```diff
@@ -139,7 +140,24 @@ class Editor extends React.PureComponent<Props, State> {
     if (params[params.length - 1] === 'view') {
       this.setState({ fullscreen: true });
     }
+
+    window.addEventListener(
+      'popstate',
```
Comment (domoritz) [inline src/components/renderer/renderer.tsx]: What is this event and why can't we move this code where this event is triggered?

Comment (siddhant1) [reply src/components/renderer/renderer.tsx]: This event is triggered on back button press and that is handled internally by browser

Comment (domoritz) [reply src/components/renderer/renderer.tsx]: But doesn't react route listen to this already? Can't we use that?

File: src/components/renderer/renderer.tsx [hunk 3]
```diff
@@ -23,6 +24,7 @@ type State = Readonly<typeof defaultState>;
 class Editor extends React.PureComponent<Props, State> {
   public static pathname: string;
   public readonly state: State = defaultState;
+  public unlisten: any;
```
Comment (domoritz) [inline src/components/renderer/renderer.tsx]: Please add a type

Comment (siddhant1) [reply src/components/renderer/renderer.tsx]: Do we have a type for it? Should I just use `() => void`

Comment (siddhant1) [reply src/components/renderer/renderer.tsx]: Also , we do not use @types/react-router-dom in the app

Comment (domoritz) [reply src/components/renderer/renderer.tsx]: Please include it then.

File: src/components/renderer/renderer.tsx (hunk 2 above)
Comment (domoritz) [reply src/components/renderer/renderer.tsx]: I added it. Please update the types.

File: src/components/renderer/renderer.tsx (hunk 3 above)
Comment (domoritz) [reply src/components/renderer/renderer.tsx]: I added it to master.

**Pull request merged**